quantmsrescore/ms2pip.py

```
@@ -1,19 +1,176 @@
+
+        # Validate number of results
+        if not results:
+            raise exceptions.NoMatchingSpectraFound(
+                "No spectra matching spectrum IDs from PSM list could be found in provided file."
+            )
+        logger.debug(f"Gathered data for {len(results)} PSMs.")
+
+        # Add XGBoost predictions if required
+        if (
+                not (vector_file or annotations_only)
+                and "xgboost_model_files" in MODELS[self.model].keys()
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)